### PR TITLE
Add readme

### PR DESCRIPTION
**Pull request by erwinc1**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+### Maven
+[![Maven Central](https://maven-badges.herokuapp.com/maven-central/io.github.rabobank.shadow_tool/shadow-tool-spring-boot-starter/badge.svg)](https://maven-badges.herokuapp.com/maven-central/io.github.rabobank.shadow_tool/shadow-tool-spring-boot-starter)
+
+```xml
+<dependency>
+    <groupId>io.github.rabobank.shadow_tool</groupId>
+    <artifactId>shadow-tool-spring-boot-starter</artifactId>
+    <version>[check maven central for latest version]</version>
+</dependency>
+```
```
Comment (martinvisser) [inline README.md:17]: Let's extend it with how to use it too. What kind of properties you can set and how to configure those Shadow Flow beans would certainly be something for this starter, not for the tool itself.

File: README.md [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+    flow1:
+      percentage: 50
+    flow2:
+      percentage: 75
+  encryption:
+    cipher:
+      secret: "3d7e0c4f8fbbd8d8a79e76cabc8f4e24"
+      initialization-vector: "3d7e0c4f8fbb"
+    public-key: "MIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEArmkP2CgDn3OsuIj1GxM3"
+    noop: false
```
Comment (martinvisser) [inline README.md:111]: Would drop this last line as combining `noop` wouldn't make any sense

File: README.md [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+This exposes bean `noopEncryptionService` of type `EncryptionService`
+```yaml
+shadowflow:
+  encryption:
+    noop: true
+```
+
+## Example Configuration
+
+Example `application.yml` configuration for a shadow flow with all possible values.:
```
Comment (martinvisser) [inline README.md:97]:
```suggestion
Example `application.yml` configuration for a shadow flow with all possible values:
```

File: README.md (hunk 2 above)
Comment (martinvisser) [reply README.md:111]: Actually combining two different encryption types will create an exception if I'm not mistaken, so configuring both a Cipher and Public Key isn't possible, but please confirm.

**Review by martinvisser — CHANGES_REQUESTED** [review]

File: README.md (hunk 2 above)
Comment (martinvisser) [reply README.md:111]: Not sure, but is it clear that `noop` is different than none?

File: README.md [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+    encryption:
+        cipher:
+            secret: "3d7e0c4f8fbbd8d8a79e76cabc8f4e24"
+            initialization-vector: "3d7e0c4f8fbb"
+```
+
+Public key example: exposes bean `publicKeyEncryptionService` of type `EncryptionService`
+```yaml 
+shadowflow:
+    encryption:
```
Comment (aaukhatov) [inline README.md:80]: I think YAML formatting is broken here, I know it's not important in README. But people can copy-paste it...

Comment (martinvisser) [reply README.md:80]: What do you mean? What is broken?
https://github.com/rabobank/shadow-tool-spring-boot-starter/blob/feat/readme/README.md looks okay

Comment (martinvisser) [reply README.md:80]: Well, `cipher` should be removed no I had another look.

Comment (martinvisser) [reply README.md:80]: There's a bit of inconsistency in the number of space between the code blocks, but that doesn't make the yaml itself invalid indeed.

File: README.md [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+| shadowflow.encryption.public-key                   | java.lang.String                                                                                                      | ""            | Base 64 encoded version of an X509 Public Key. Used in a Cipher with algorithm "RSA/ECB/OAEPWITHSHA-256ANDMGF1PADDING".                         |
+| shadowflow.encryption.noop                         | java.lang.Boolean                                                                                                     | false         | Disables encryption but encodes differences as Base64.                                                                                          |
+| shadowflow.flows[*].percentage                     | java.lang.Integer                                                                                                     | 0             | Percentage of how many calls should be compared in the shadow flow. Should be in the range of 0-100. Zero effectively disables the shadow flow. |
```
Comment (martinvisser) [inline README.md:35]: Think these will be more readable if wrapped in back-ticks. But moreover, I think we can ditch the type, or at least simplify them a lot. For example, nobody cares it's actually a `ShadowFlowConfig`, it's just a map of Shadow Flows. Same about CipherProperties, it's irrelevant: the nested fields are relevant, not the "root".

And you can remove things like `java.lang` here too IMO.

IMO everything that looks like code or a script should be wrapped with back-ticks. e.g `RSA/ECB/OAEPWITHSHA-256ANDMGF1PADDING` and `openssl rand -hex 12`

File: README.md [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+Public key example: exposes bean `publicKeyEncryptionService` of type `EncryptionService`
+```yaml 
+shadowflow:
+    encryption:
+        cipher:
+          public-key: "MIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEArmkP2CgDn3OsuIj1GxM3"
+```
+
+3. If you want to disable encryption but encode differences as Base64, set the shadowflow.encryption.noop property to
+   true.
```
Comment (martinvisser) [inline README.md:86]:
```suggestion
3. If you want to disable encryption but encode differences as Base64, set the `shadowflow.encryption.noop` property to
   `true`.
```

File: README.md (hunk 4 above)
Comment (aaukhatov) [reply README.md:80]: I meant indents

**Review by martinvisser — APPROVED** [review]

**Pull request merged**